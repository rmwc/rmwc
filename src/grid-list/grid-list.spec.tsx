import React from 'react';
import { render } from '@testing-library/react';
import {
  GridList,
  GridTile,
  GridTileIcon,
  GridTilePrimary,
  GridTilePrimaryContent,
  GridTileSecondary,
  GridTileTitle
} from './';

describe('GridList', () => {
  it('renders', () => {
    const { asFragment } = render(
      <GridList>
        <GridTile>
          <GridTilePrimary>
            <GridTilePrimaryContent src="" alt="" />
          </GridTilePrimary>
          <GridTileSecondary>
            <GridTileIcon icon="icon" />
            <GridTileTitle>Tile</GridTileTitle>
          </GridTileSecondary>
        </GridTile>
      </GridList>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('can be tileGutter1', () => {
    const { container } = render(<GridList tileGutter1 />);
    expect(container.firstChild).toHaveClass('mdc-grid-list--tile-gutter-1');
  });

  it('can be headerCaption', () => {
    const { asFragment } = render(<GridList withIconAlignStart />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('can be headerCaption', () => {
    const { container } = render(<GridList headerCaption />);
    expect(container.firstChild).toHaveClass('mdc-grid-list--header-caption');
  });

  it('can be tileAspect 1x1', () => {
    const { container } = render(<GridList tileAspect="1x1" />);
    expect(container.firstChild).toHaveClass('mdc-grid-list--tile-aspect-1x1');
  });

  it('can be tileAspect undefined', () => {
    const { asFragment } = render(<GridList tileAspect={undefined} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('can have custom classnames', () => {
    [
      GridList,
      GridTile,
      GridTileIcon,
      GridTilePrimary,
      GridTileSecondary,
      GridTileTitle
    ].forEach((Component: React.ComponentType<any>) => {
      const { container } = render(
        <Component className={'my-custom-classname'} />
      );
      expect(container.firstChild).toHaveClass('my-custom-classname');
    });
  });
});
