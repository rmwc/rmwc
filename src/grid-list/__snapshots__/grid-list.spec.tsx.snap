// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridList can be headerCaption 1`] = `
<DocumentFragment>
  <div
    class="mdc-grid-list mdc-grid-list--with-icon-align-start mdc-grid-list--tile-aspect-1x1"
  >
    <ul
      class="mdc-grid-list__tiles"
    />
  </div>
</DocumentFragment>
`;

exports[`GridList can be tileAspect undefined 1`] = `
<DocumentFragment>
  <div
    class="mdc-grid-list mdc-grid-list--tile-aspect-1x1"
  >
    <ul
      class="mdc-grid-list__tiles"
    />
  </div>
</DocumentFragment>
`;

exports[`GridList renders 1`] = `
<DocumentFragment>
  <div
    class="mdc-grid-list mdc-grid-list--tile-aspect-1x1"
  >
    <ul
      class="mdc-grid-list__tiles"
    >
      <li
        class="mdc-grid-tile"
      >
        <div
          class="mdc-grid-tile__primary"
        >
          <img
            alt=""
            class="mdc-grid-tile__primary-content"
            src=""
          />
        </div>
        <span
          class="mdc-grid-tile__secondary"
        >
          <i
            class="rmwc-icon rmwc-icon--ligature material-icons mdc-grid-tile__icon"
          >
            icon
          </i>
          <span
            class="mdc-grid-tile__title"
          >
            Tile
          </span>
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
